### PR TITLE
Slight refactoring of PINNs

### PR DESCRIPTION
**Pull request by KirillZubov**
NNPDEProblem -> OptimizationProblem

### CODE REVIEW

File: src/pinns_pde_solve.jl [hunk 1]
```diff
@@ -293,35 +293,69 @@ function generator_training_sets(domains, discretization, bound_args, dict_indva
...
+    bc_funcs = eval.(extract_bc_funcs)
 
     # generate training sets
     train_sets = generator_training_sets(domains, discretization, bound_args, dict_indvars)
 
-    return NNPDEProblem(_pde_func,_bc_funcs, train_sets, dim)
+    # get loss_function
+    loss_function = get_loss_function(pde_func,bc_funcs,train_sets)
+
+	return GalacticOptim.OptimizationProblem(loss_function, zeros(dim), p=nothing)
```
Comment (Vaibhavdixit02) [inline src/pinns_pde_solve.jl:354]: Not an issue but not passing in `p` as `nothing` should work here

**Comment by Vaibhavdixit02** [comment]
Not sure if `OptimizationFunction` could be used here, where do you keep the derivatives of the loss function?

File: src/pinns_pde_solve.jl (hunk 1 above)
Comment (KirillZubov) [reply src/pinns_pde_solve.jl]: yes, thanks

**Comment by KirillZubov** [comment]
> Not sure if `OptimizationFunction` could be used here, where do you keep the derivatives of the loss function?

As long as we use the numeric derivative, this is not necessary, but further to support AD it will probably be convenient.

**Comment by KirillZubov** [comment]
@Vaibhavdixit02 also, I have the problem with add GalactiOptim. We get the error. 
`ERROR: Unknown package Zygote found in the compatibility requirements of GalacticOptim `

**Comment by Vaibhavdixit02** [comment]
@KirillZubov that doesn't look good, can you share your `versioninfo`?  @abhigupta768 have you seen this?

**Comment by KirillZubov** [comment]
@Vaibhavdixit02 
```
julia> versioninfo()
Julia Version 1.4.2
Commit 44fa15b150* (2020-05-23 18:35 UTC)
Platform Info:
  OS: macOS (x86_64-apple-darwin18.7.0)
  CPU: Intel(R) Core(TM) i5-3427U CPU @ 1.80GHz
  WORD_SIZE: 64
  LIBM: libopenlibm
  LLVM: libLLVM-8.0.1 (ORCJIT, ivybridge)
Environment:
  JULIA_EDITOR = atom  -a
  JULIA_NUM_THREADS = 2
  JULIA_NUM_THREADS = 2
```

**Comment by KirillZubov** [comment]
@Vaibhavdixit02  @ChrisRackauckas, I copied the OptimizationProblem and removed the dependence on the GalacticOptim for a while until the problem with add GalacticOptim will solve.

**Comment by abhigupta768** [comment]
I can not recreate the error with master, it's probably an issue with the release. @KirillZubov can you check with the master for your use?

**Comment by KirillZubov** [comment]
@abhigupta768 yes, it works: `add GalacticOptim#master`

**Comment by ChrisRackauckas** [comment]
let's do a 0.2 tag today then

**Comment by ChrisRackauckas** [comment]
The issue was something in the registry: https://github.com/JuliaRegistries/General/pull/19399

**Comment by ChrisRackauckas** [comment]
This isn't quite there yet because this package should not define a solve for an optimization problem, but it's moving in the right direction.

**Pull request merged**